ArtOfIllusion/src/artofillusion/script/ExecuteScriptWindow.java

```diff
@@ -168,66 +355,121 @@ private void saveScript()
         new BStandardDialog(null, new String [] {Translate.text("errorWritingScript"),
           ex.getMessage() == null ? "" : ex.getMessage()}, BStandardDialog.ERROR).showMessageDialog(this);
       }
-      setScriptNameFromFile(fc.getSelectedFile().getName());
-  
+      // Now we have saved, we can't change the language
+      languageChoice.setEnabled(false);
+      updateEditableStatus(scriptPath, fc.getSelectedFile().getAbsolutePath());
+      scriptPath = fc.getSelectedFile().getAbsolutePath();
+
+      setScriptNameFromFile(fc.getSelectedFile().getAbsolutePath());
       // Update the Scripts menus in all windows.
-      EditingWindow allWindows[] = ArtOfIllusion.getWindows();
-      for (int i = 0; i < allWindows.length; i++)
-        if (allWindows[i] instanceof LayoutWindow)
-          ((LayoutWindow) allWindows[i]).rebuildScriptsMenu();
+        for (EditingWindow edWin : ArtOfIllusion.getWindows())
+        {
+            if (edWin instanceof LayoutWindow)
+            {
+                ((LayoutWindow) edWin).rebuildScriptsMenu();
+            }
+        }
```

Comment:
Hope this doesn't sound too nit-picky, but a bit of a style comment:

We generally don't use braces for one-line `if (condition) doThis;` statements.

I also see that at least some of your additions are indented 3 spaces, rather than the 2 that we try for. (I am aware that not all of our existing code matches this.) Is this something that we can make easier to set in an IDE's auto-formatter?

Reply:
I don't use braces for one-ifs either ; this is a mistake, thanks for pointing it out and for the reminder about the 2 spaces. I'll check my Netbeans formatting settings.